src/resolver.ts

```diff
@@ -5,23 +5,70 @@ import { CustomError } from './errors';
 import { LonginInput, UserInput } from './schema-types';
 import { sign, verify } from 'jsonwebtoken';
 
+function verifyToken(context) {
+  if (!context.token) {
+    throw new CustomError('Voce nao tem permissao para efetuar essa açao', 404, 'token not found');
+  }
+
+  if (!verify(context.token, 'supersecret')) {
+    throw new CustomError('Ação nao autorizada', 401);
+  }
+}
+
 export const resolvers = {
   Query: {
+    user: async (_, { id }, context) => {
+      verifyToken(context);
+
+      const repository = getRepository(User);
+      const resp = await repository.findOne({ id });
+
+      if (!resp) {
+        throw new CustomError('Usuário não encontrado', 404);
+      }
+      return resp;
+    },
+
     hello: (): string => {
       return 'hello world';
     },
-  },
-  Mutation: {
-    createUser: async (_, { data: args }: { data: UserInput }, context) => {
-      if (!context.token) {
-        throw new CustomError('Voce nao tem permissao para efetuar essa açao', 404, 'token not found');
+
+    users: async (_: string, { quantity, page }, context) => {
+      verifyToken(context);
+
+      let pageBefore = false;
+      let pageAfter = false;
+      let shown = '';
+
+      const repository = getRepository(User);
+
+      quantity == undefined ? (quantity = 10) : quantity;
+      page == undefined ? (page = 0) : (page += -1);
+      const skip = page * quantity;
+
+      const list = await repository.createQueryBuilder().orderBy('name').limit(quantity).offset(skip).getMany();
+
+      if (page > 0) {
+        pageBefore = true;
+      }
+      if ((await repository.findOne({ id: skip + quantity + 1 })) != undefined) {
+        pageAfter = true;
```

Comment:
Essa lógica pode até dar certo pra esse estado do banco em específico, mas é uma lógica um pouco perigosa porque nem sempre os ids serão certinhos de 1 a X. Se um usuário for excluído, por exemplo, já ferra essa lógica, certo?

Como vc pode resolver isso de forma mais consistente?